RabbitMQ.Stream.Client/Connection.cs

```diff
@@ -85,41 +86,28 @@ private static System.IO.Stream MaybeTcpUpgrade(NetworkStream networkStream, Ssl
             return new Connection(socket, commandCallback, closedCallBack, sslOption);
         }
 
-        private ValueTask<FlushResult> FlushAsync()
-        {
-            lock (writeLock)
-            {
-                var flushTask = writer.FlushAsync();
-                return flushTask;
-            }
-        }
-
         public async ValueTask<bool> Write<T>(T command) where T : struct, ICommand
         {
-            WriteCommand(command);
-            var flushTask = FlushAsync();
-
-            // Let's check if this is completed synchronously before invoking the async state machine
-            if (!flushTask.IsCompletedSuccessfully)
-            {
-                await flushTask.ConfigureAwait(false);
-            }
-
-            return flushTask.Result.IsCompleted;
+            await WriteCommand(command);
+            // we return true to indicate that the command was written
+            return true;
         }
 
-        private void WriteCommand<T>(T command) where T : struct, ICommand
+        private async Task WriteCommand<T>(T command) where T : struct, ICommand
```

Comment:
This is the main change. 
It makes `WriteCommand ` async using `WriteAsync` Add `FlushAsync`  instead of [sync wait the task](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/220/files#diff-71215640c5968601ad0a124946241692beb3c931cc450bb43bb7061e8f84c47eL245)